lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -961,6 +965,30 @@ def set_http_request(record, entry)
       output.cache_validated_with_origin_server = \
         cache_validated_with_origin_server \
         unless cache_validated_with_origin_server.nil?
+
+      latency = input.delete('latency')
+      unless latency.nil?
+        # Parse latency. If no valid format is detected, skip setting latency.
+        # Format: whitespace (optional) + integer + point & decimal (optional)
+        #       + whitespace (optional) + "s" + whitespace (optional)
+        # e.g.: "1.42 s"
+        match = @http_latency_regexp.match(latency)
+        if match
+          # Split the integer and decimal parts in order to calculate seconds
+          # and nanos.
+          latency_seconds = match['seconds'].to_i
+          latency_nanos = (match['decimal'].to_f * NANOS_IN_A_SECOND).round
+          if @use_grpc
+            output.latency = Google::Protobuf::Duration.new
+          else
+            output.latency = {}
+          end
+          output.latency['seconds'] = latency_seconds \
```

Comment:
This will happen automagically for protobufs, so I would just simplify the hash case:
```
            output.latency = {
              seconds: latency_seconds,
              nanos: latency_nanos
            }.delete_if { |k, v| v == 0 }
```